src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherState.scala

```diff
@@ -68,26 +67,33 @@ case class BlockFetcherState(
       .orElse(waitingHeaders.headOption.map(_.number))
       .getOrElse(lastBlock)
 
-  def nextToLastBlock: BigInt = lastBlock + 1
+  def nextToLastBlock: BigInt = waitingHeaders.lastOption
```

Comment:
Should we rename this as blockHeaderToFetch or something like that? (also adding some comments would be nice)